server_src/lib/dataFunctions/printerClean.js

```diff
@@ -61,8 +61,15 @@ class PrinterClean {
    * @deprecated Use cache/printer.cache.js instead
    * @returns {[]}
    */
-  static returnPrintersInformation() {
-    return printersInformation;
+  static returnPrintersInformation(id) {
+    if (!id) {
+      return printersInformation;
+    } else {
+      const index = _.findIndex(printersInformation, function (o) {
+        return o._id == id;
+      });
+      return printersInformation[index];
+    }
```

Comment:
Updated this function to work with and without a specific printer. No id will return the entire list, and id provided will pull out the requested printer. This will simplify the code elseware a lot with the check done here. Follow up PR to update to utilise this more but will await the cache finishing for printers.

Reply:
Im not a fan of this code style as the function becomes indeterministic. Simpler functions have simpler bugs.
Usually I create a 'getById' and a 'list' variant to be 100% clear, also because this is the CRUD approach (get, list are separated). Can you point to the code which becomes much simpler, help me see the impact a bit?

Btw, of course sometimes ease of use is really nice. I fully understand that 😉. Just wanted to let you know that in such cases either the code which uses this function could benefit from unit tests to make absolutely sure the `id` parameter is not left null/false/empty/0/NaN. If I see those test, you would not have heard me ring a bell as I would know you'd tested it and your thus in full control. 
Those tests are probably high in count, so splitting it up is: 1) less impactful 2) really clear 3) less code to cover in tests

Reply:
![image](https://user-images.githubusercontent.com/6005355/119563337-af1c7d00-bda7-11eb-9a1c-dbf4495dd811.png)
Changing the function to 2 types doesnt seem like it will impact the code.